tools/configure

```diff
@@ -1,291 +1,167 @@
+                        help="The format of Macros file to generate. If "
+                        "'Makefile' is passed in, a file called 'Macros.make' "
+                        "is generated. If 'CMake' is passed in, a file called "
+                        "'Macros.cmake' is generated. This option can be "
+                        "specified multiple times to generate multiple files. "
+                        "If not used at all, Macros generation is skipped. "
+                        "Note that Depends files are currently always in "
+                        "Makefile format, regardless of this option.")
+    parser.add_argument("--output-dir", default=os.getcwd(),
+                        help="The directory to write files to. If not "
+                        "specified, defaults to the current working directory.")
+
+    args = parser.parse_args()
+    CIME.utils.handle_standard_logging_options(args)
+
+    opts = {}
+    if args.machines_dir is not None:
+        machines_file = os.path.join(args.machines_dir, "config_machines.xml")
+        machobj = Machines(infile=machines_file, machine=args.machine)
+    else:
+        if os.environ.get('CIME_MODEL') is not None:
+            machobj = Machines(machine=args.machine)
+        else:
+            expect(False, "Either --mach-dir or the CIME_MODEL environment "
+                   "variable must be specified!")
+
+    opts['machobj'] = machobj
+
+    if args.macros_format is None:
+        opts['macros_format'] = []
+    else:
+        opts['macros_format'] = args.macros_format
+
+    expect(os.path.isdir(args.output_dir),
+           "Output directory '%s' does not exist." % args.output_dir)
+
+    opts['output_dir'] = args.output_dir
+
+    # Set compiler.
+    if "COMPILER" in os.environ:
+        compiler = os.environ["COMPILER"]
+    else:
+        compiler = machobj.get_default_compiler()
+        os.environ["COMPILER"] = compiler
+    expect(opts['machobj'].is_valid_compiler(compiler),
+           "Invalid compiler vendor given in COMPILER environment variable: %s"
+           % compiler)
+    opts['compiler'] = compiler
+
+    # Set MPI library.
+    if "MPILIB" in os.environ:
+        mpilib = os.environ["MPILIB"]
+    else:
+        mpilib = machobj.get_default_MPIlib()
+        os.environ["MPILIB"] = mpilib
+    expect(opts['machobj'].is_valid_MPIlib(mpilib),
+           "Invalid MPI library name given in MPILIB environment variable: %s" %
+           mpilib)
+    opts['mpilib'] = mpilib
+
+    # Set DEBUG flag.
+    if "DEBUG" in os.environ:
+        expect(os.environ["DEBUG"].lower() in ('true', 'false'),
+               "Invalid DEBUG environment variable value (must be 'TRUE' or "
+               "'FALSE'): %s" % os.environ["DEBUG"])
+        debug = os.environ["DEBUG"].lower() == "true"
+    else:
+        debug = False
+        os.environ["DEBUG"] = "FALSE"
+    opts['debug'] = debug
+
+    return opts
+
+def _main():
+    opts = parse_command_line(sys.argv)
```

Comment:
Consider making this functionality a library so that others can access without having to issue shell command.